sormas-api/src/main/java/de/symeda/sormas/api/report/AggregatedCaseCountDto.java

```diff
@@ -291,39 +237,181 @@ public void setEpiWeek(Integer epiWeek) {
+			labConfirmations,
+			deaths,
+			ageGroup);
+	}
+
+	public boolean similar(Object o, AggregateReportGroupingLevel groupingLevel) {
+
+		if (this == o)
+			return true;
+		if (o == null || getClass() != o.getClass())
 			return false;
-		if (labConfirmations != other.labConfirmations)
+		AggregatedCaseCountDto that = (AggregatedCaseCountDto) o;
+		boolean similar = year == that.year && epiWeek == that.epiWeek && disease == that.disease && Objects.equals(ageGroup, that.ageGroup);
+		if (groupingLevel != null) {
+			switch (groupingLevel) {
+
+			case REGION:
+				similar = similar && Objects.equals(regionName, that.regionName) && Objects.equals(regionId, that.regionId);
+				break;
+			case DISTRICT:
+				similar = similar && Objects.equals(districtName, that.districtName) && Objects.equals(districtId, that.districtId);
+				break;
+			case HEALTH_FACILITY:
+				similar =
+					similar && Objects.equals(healthFacilityName, that.healthFacilityName) && Objects.equals(healthFacilityId, that.healthFacilityId);
+				break;
+			case POINT_OF_ENTRY:
+				similar = similar && Objects.equals(pointOfEntryName, that.pointOfEntryName) && Objects.equals(pointOfEntryId, that.pointOfEntryId);
+				break;
+			default:
+			}
+		}
+		return similar;
+	}
+
+	public boolean equalJurisdiction(Object o) {
+		if (this == o)
+			return true;
+		if (o == null || getClass() != o.getClass())
 			return false;
-		if (newCases != other.newCases)
+		AggregatedCaseCountDto that = (AggregatedCaseCountDto) o;
+		return Objects.equals(regionName, that.regionName)
+			&& Objects.equals(regionId, that.regionId)
+			&& Objects.equals(districtName, that.districtName)
+			&& Objects.equals(districtId, that.districtId)
+			&& Objects.equals(healthFacilityName, that.healthFacilityName)
+			&& Objects.equals(healthFacilityId, that.healthFacilityId)
+			&& Objects.equals(pointOfEntryName, that.pointOfEntryName)
+			&& Objects.equals(pointOfEntryId, that.pointOfEntryId);
+	}
+
+	public boolean higherJurisdictionLevel(Object o) {
+
+		if (this == o)
+			return true;
+		if (o == null || getClass() != o.getClass())
 			return false;
-		if (!Objects.equals(ageGroup, other.ageGroup))
+		AggregatedCaseCountDto that = (AggregatedCaseCountDto) o;
+
+		if (regionId == null && that.regionId != null) {
+			return true;
+		}
+		if (districtId == null && that.districtId != null) {
+			return true;
+		}
+		if (healthFacilityId == null && that.healthFacilityId != null && pointOfEntryId == null && that.pointOfEntryId == null) {
+			return true;
+		}
+		if (pointOfEntryId == null && that.pointOfEntryId != null && healthFacilityId == null && that.healthFacilityId == null) {
+			return true;
+		}
+		return false;
+	}
+
+	public boolean sameJurisdictionLevel(Object o) {
+
+		if (this == o)
+			return true;
+		if (o == null || getClass() != o.getClass())
 			return false;
-		return true;
+		AggregatedCaseCountDto that = (AggregatedCaseCountDto) o;
+
+		if (regionId == null && that.regionId == null) {
```

Comment:
What about:
```suggestion
		if (regionId == null || that.regionId == null) {
		return regionId == null && that.regionId == null;
		}
```
Thisway you can completely ignore the region below. Same for the district in the next if.

Reply:
Maybe even easier: Use the getJursidictionLevel method below and compare the results